### PR TITLE
Tests added

### PR DESCRIPTION
**Pull request by ah1053**
1. Added Tests that covers (80% of the code)
2. Changed rooms , available rooms and full rooms to  a Map DataType adviced by @franklin-huang-mobiledgex  

> However the client will still receive an array to avoid serialization problems on the C# side of things.
> Room.roomMembers will still remain an array to avoid nested serialization problems on C#

3. Changed example.js to app.js as adviced by @lgarner 
4. Deleted ClientExamples, now they are all used in test/main_test.js.

Screenshot of main_test coverage using nyc
Screen Shot 2021-04-27 at 10 20 19 AM

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -6,7 +6,7 @@ COPY package*.json /usr/src/app/
 RUN npm install
 COPY . /usr/src/app
 
-CMD node example.js
+CMD node app.js
 
 EXPOSE 3000
 EXPOSE 5000/udp
```
Comment (franklin-huang-mobiledgex) [inline Dockerfile:12]: nit newline

File: models/Lobby.js [hunk 2]
```diff
@@ -33,9 +33,9 @@ class Lobby {
         this.MAX_ROOMS_PER_LOBBY = MAX_ROOMS_PER_LOBBY
         this.connectedClients = {}
         this.nameMap = {}
-        this.rooms = []
-        this.availableRooms = []
-        this.fullRooms = []
+        this.rooms = new Map()
+        this.availableRooms = new Map()
+        this.fullRooms = new Map()
```
Comment (franklin-huang-mobiledgex) [inline models/Lobby.js:38]: thanks!

File: test/main_test.js [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+          setTimeout(() => 
+          { 
+            expect(server.edgeMultiplay.lobby.rooms.get(roomId).udpConnections.size).equal(1) 
+            done()
+          }, 100);
+          break
+      }
+    }
+  })
+})
```
Comment (franklin-huang-mobiledgex) [inline test/main_test.js:161]: nit: new line

File: test/main_test.js [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+  it('Test create room', (done) => {
+    var ws = new WebSocket('ws://localhost:3000');
+    ws.onopen = function () {
+      console.log('client 1 is open')
+    }
+    ws.onmessage = function (e) {
+      jsonObj = JSON.parse(e.data);
+      switch (jsonObj.type) {
+        case 'register':
+          expect(jsonObj.playerId).to.be.string
```
Comment (franklin-huang-mobiledgex) [inline test/main_test.js:17]: do you need a check for what value the playerId is (eg. expect(jsonObj.playerId).equal("Ahmed") or something)?

File: test/main_test.js [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+        case 'notification':
+          expect(jsonObj.notificationText).equal('new-room-created-in-lobby')
+          break
+
+        case 'roomCreated':
+          roomId = jsonObj.room.roomId
+          expect(roomId).to.be.string
+          expect(server.edgeMultiplay.lobby.rooms.size).equal(1)
+          expect(server.edgeMultiplay.lobby.availableRooms.size).equal(1)
+          expect(server.edgeMultiplay.lobby.fullRooms.size).equal(0)
```
Comment (franklin-huang-mobiledgex) [inline test/main_test.js:37]: it looks like some of these checks are repeated (lines 72-74 and other places). may want a helper function

**Review by franklin-huang-mobiledgex — APPROVED** [review]
Great! Thanks for switching those over to maps. Test cases are great!

File: test/main_test.js (hunk 4 above)
Comment (ah1053) [reply test/main_test.js:17]: Unfortunately playerId is a unique id

File: Dockerfile (hunk 1 above)
Comment (ah1053) [reply Dockerfile:12]: the extra line here is intentional as a way to show the main file.

File: models/Lobby.js (hunk 2 above)
Comment (ah1053) [reply models/Lobby.js:38]: No, Thank you

File: Dockerfile (hunk 1 above)
Comment (franklin-huang-mobiledgex) [reply Dockerfile:12]: adding a newline at the end of the file will get rid of that red circle warning "No newline at end of file"

Comment (ah1053) [reply Dockerfile:12]: Aha thanks

File: test/main_test.js (hunk 5 above)
Comment (ah1053) [reply test/main_test.js:37]: Yeah those are redundant, Thanks for catching that.

**Pull request merged**